src/parser.rs

```diff
@@ -99,6 +106,7 @@ impl Phonetic {
                             Some(rule) => output.push_str(rule.replace_with),
                             None => output.push_str(default_replacement),
                         };
+                        output.push_str(extra);
```

Comment:
As this push is being done for both if and else branches, it can be outside branching.